fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java

```diff
@@ -365,6 +366,53 @@ public Channel getChannel(HFClient hfClient, String channelName, User fabricUser
         return channel;
     }
 
+    private Channel.PeerOptions getPeerOptionsWithRoles(String channelName, String peerName, Channel.PeerOptions commonPeerOptions) {
+
+        //Add all roles by default except ServiceDiscovery in line with NetworkConfig behaviour
+        EnumSet<Peer.PeerRole> peerRoles = EnumSet.of(Peer.PeerRole.CHAINCODE_QUERY,
+                Peer.PeerRole.ENDORSING_PEER,
+                Peer.PeerRole.LEDGER_QUERY,
+                Peer.PeerRole.EVENT_SOURCE);
+
+        Map<String, Boolean> roles = getRoot().channels.get(channelName).peers.get(peerName).roles;
+        if (roles != null) {
+            roles.forEach((key, value) -> {
+                // Remove explicitly forbidden roles
+                if (!value) {
+                    peerRoles.remove(getPeerRoleByName(key));
+                }
+                //Add ServiceDiscovery role if it is explicitly enabled
+                if (key.equals(Peer.PeerRole.SERVICE_DISCOVERY.getPropertyName()) && value) {
+                    peerRoles.add(Peer.PeerRole.SERVICE_DISCOVERY);
+                }
+            });
```

Comment:
It seems to be pretty strange handling. It is possible to specify a set, but these are completely ignored. If there any way of handling this better?
For example  - if a `peers` set contains only 'false' values. start from the set above. if there any `true` entries, start from the `true` set. Additionally we can handle 'ALL' special property name.
Another option would be support `custom` top-level entry and use yaml anchors to drop in customized set. What do you think?

Reply:
As discussed changed default values for all roles for true